tests/unit/io/Cereal_unittest.cc

```diff
@@ -0,0 +1,174 @@
+#include <shogun/lib/common.h>
+#include <gtest/gtest.h>
+#include "CerealObject.h"
+
+#include <shogun/lib/SGVector.h>
+#include <cereal/archives/json.hpp>
```

Comment:
why do you need this?
or rather, why dont we use the `save_json` and `load_json` methods?